linkerd/app/outbound/src/lib.rs

```diff
@@ -272,27 +268,72 @@ impl Config {
             .push_spawn_ready()
             .check_service::<Target<HttpEndpoint>>()
             .push(discover)
-            .push_on_response(http::balance::layer(EWMA_DEFAULT_RTT, EWMA_DECAY))
+            .push_on_response(
+                svc::layers()
+                    .push(http::balance::layer(EWMA_DEFAULT_RTT, EWMA_DECAY))
+                    // If the balancer has been empty/unavailable for 10s, eagerly fail
+                    // requests.
+                    .push_failfast(dispatch_timeout)
+                    .push(metrics.stack.layer(stack_labels("balance"))),
+            )
             .into_new_service()
+            .instrument(|c: &Concrete<http::Settings>| info_span!("balance", addr = %c.addr))
+            .check_new_service::<Concrete<http::Settings>, http::Request<_>>();
+
+        // Routes `Logical` targets to a cached `Profile` stack, i.e. so that profile
+        // resolutions are shared even as the type of request may vary.
```

Comment:
again, maybe worth a quick explanation of how the receiver/addr moves through the stack here?